plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE LambdaCase #-}
+{-|
+A simple beta-reduction pass.
+
+-}
+module PlutusIR.Transform.Beta (
+  beta
+  ) where
+
+import           PlutusPrelude
+
+import           PlutusIR
+
+{-|
+A single non-recursive application of the beta rule.
+
+- TODO: check if annotations are applied correctly, and
+
+-}
+
+-- Local beta transformation
+betaStep
+    :: Term tyname name uni fun a
+    -> Term tyname name uni fun a
+betaStep = \case
+    Apply a (LamAbs _ name typ body) arg ->
+        let varDecl  = VarDecl a name typ
+            binding  = TermBind a Strict varDecl arg
+            bindings = binding :| []
+        in
+            Let a NonRec bindings body
+    t -> t
```

Comment:
Since you brought it up: you might as well handle immediately instantiated type abstractions here too.

Reply:
How should handling immediately instantiated type abstractions work?

```
    (/\ a : K. (\x : a). x) {A}
==> let a : K = A in (\x : a) . x
```
?

Reply:
This seems to be blocked by SCP-2141.  I added SCP-2152 to track this.

Reply:
> How should handling immediately instantiated type abstractions work?

Substitute `A` for `a` everywhere in the body?  The term above would get erased to 
```
(force (delay (lam x x)))
```
which should be optimised to `(lam x x)`, so whatever we do should respect that.

Reply:
I think what Jakub said is right, we can just turn it into a let-binding here, given that that's what we're doing for value abstractions.